cmake/modules/boards.cmake

```diff
@@ -92,11 +92,28 @@ if(DEFINED ZEPHYR_BOARD_ALIASES)
     set(BOARD_IDENTIFIER ${BOARD_ALIAS_IDENTIFIER}${BOARD_IDENTIFIER})
   endif()
 endif()
+
 include(${ZEPHYR_BASE}/boards/deprecated.cmake)
 if(${BOARD}_DEPRECATED)
   set(BOARD_DEPRECATED ${BOARD} CACHE STRING "Deprecated board name, provided by user")
-  set(BOARD ${${BOARD}_DEPRECATED})
+  parse_board_components(${BOARD}_DEPRECATED BOARD BOARD_DEPRECATED_REVISION BOARD_DEPRECATED_IDENTIFIER)
   message(WARNING "Deprecated BOARD=${BOARD_DEPRECATED} name specified, board automatically changed to: ${BOARD}.")
+  if(DEFINED BOARD_DEPRECATED_REVISION)
+    if(DEFINED BOARD_REVISION)
+      message(FATAL_ERROR
+        "Board revision '${BOARD_REVISION}' conflicts with the new board's revision: "
+        "${BOARD}@${BOARD_DEPRECATED_REVISION}, corresponding to '${BOARD_DEPRECATED}'."
+      )
+    endif()
+    set(BOARD_REVISION ${BOARD_DEPRECATED_REVISION})
+  endif()
+  if(DEFINED BOARD_IDENTIFIER)
+    message(FATAL_ERROR
+      "Deprecated boards cannot have board identifiers: ${BOARD_DEPRECATED}${BOARD_IDENTIFIER}.\n"
+      "Please specify board without an identifier.\n"
```

Comment:
could you please create a followup task, cause i'm pretty sure we're gonna need support for this in future.

Especially when someone comes along a deems one of the variant names chosen during v2 porting should be adjusted.